File: Server/src/main/java/fr/intech/roomies/model/finance/Category.java
package fr.intech.roomies.model.finance;

import fr.intech.roomies.model.buiding.Coloc;

import javax.persistence.*;
import java.util.List;

@Entity
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int categoryId;

    @Column(nullable = false)
    private String categoryName;

    @Column(nullable = false)
    private String pic;

    @ElementCollection
    private List<Budget> budgets;

    public Category(){}

    public int getCategoryId() {
        return categoryId;
    }

    public void setCategoryId(int categoryId) {
        this.categoryId = categoryId;
    }

    public String getCategoryName() {
        return categoryName;
    }

    public void setCategoryName(String categoryName) {
        this.categoryName = categoryName;
    }

    public String getPic() {
        return pic;
    }

    public void setPic(String pic) {
        this.pic = pic;
    }

    public List<Budget> getBudgets() {
        return budgets;
    }

    public void setBudgets(List<Budget> budgets) {
        this.budgets = budgets;
    }
}

File: Roomies2.0/src/Roomies2.DAL/Model/Finance/Category.cs
#region

using System;
using System.Collections.Generic;

#endregion

namespace Roomies2.DAL.Model.Finance
{
    public class Category
    {
        public Category(int categoryId = default, string categoryName = null, string categoryPicture = null,
            List<Budget> budgets = null)
        {
            CategoryId = categoryId;
            CategoryName = categoryName ?? throw new ArgumentNullException(nameof(categoryName));
            CategoryPicture = categoryPicture ?? throw new ArgumentNullException(nameof(categoryPicture));
            Budgets = budgets ?? throw new ArgumentNullException(nameof(budgets));
        }

        public int CategoryId { get; set; }
        public string CategoryName { get; set; }
        public string CategoryPicture { get; set; }
        public List<Budget> Budgets { get; set; }
    }
}